nbconvert/exporters/script.py

```diff
@@ -19,15 +18,29 @@ def _template_file_default(self):
 
     def from_notebook_node(self, nb, resources=None, **kw):
         langinfo = nb.metadata.get('language_info', {})
-        
         # delegate to custom exporter, if specified
         exporter_name = langinfo.get('nbconvert_exporter')
         if exporter_name and exporter_name != 'script':
             self.log.debug("Loading script exporter: %s", exporter_name)
-            from .export import get_exporter
+            from .export import get_exporter, ExporterNameError
             if exporter_name not in self._exporters:
-                Exporter = get_exporter(exporter_name)
-                self._exporters[exporter_name] = Exporter(parent=self)
+                try: 
+                    Exporter = get_exporter(exporter_name)
+                    self._exporters[exporter_name] = Exporter(parent=self)
+                except ValueError:
+                    if langinfo.get('nbconvert_exporter_package',""):
+                        external_exporter_name = langinfo.get('nbconvert_exporter_package')
+                        try: 
+                            Exporter = get_exporter(external_exporter_name)
```

Comment:
I don't think we should store tah in each notebook metadata, it feels to me as a recipe for confusion. This seem already that it will be used only for language specific hook, and thus I woudl much rather apply that at the script exporter lever via entry point: Language->Exporter.

First it allows us not to upgrade all notebooks metadata which is always a complicated thing.

Plus having that on a per notebook basis will lead to some notebook being converted in some ways while other won't which will be nightmare if people exchange notebook. 

it seem to me that hook into the script-exporter also make things simpler. In particular if it's in kernelspec **and** you need to have a python-package that match it's a recipe for inconsistency. Plus adding this bunch of metadate make it a de facto standard, while other tools that convert notebook might want to behave differently. 

I'm pretty sure we should even remove our own "nbconvert_exporter" key and make it custom logic in nbconvert.